### ISSUE TITLE
kornia.angle_axis_to_rotation_matrix does not seem working 

### ISSUE THREAD

**Issue by cvfish** [kornia/kornia#209]
Hi there, 

I did a pip install from source and found that 
konia.angle_axis_to_rotation_matrix and konia.rotation_matrix_to_angle_axis are not consistent.

I did the following: 

input = torch.rand(1, 3)
output = kornia.angle_axis_to_rotation_matrix(input)
input_back = kornia.rotation_matrix_to_angle_axis(output)

input and input_back will not be the same.

Also I tried quaternion, which seems working fine.

input = torch.rand(1, 3)
output = kornia.angle_axis_to_quaternion(input)
input_back = kornia.quaternion_to_angle_axis(output)
rotation_matrix_output = kornia.quaternion_to_rotation_matrix(output)
output_back = kornia.rotation_matrix_to_quaternion(rotation_matrix_output)

input_matrix_to_angle_axis = kornia.rotation_matrix_to_angle_axis(rotation_matrix_output)

input, input_back and input_matrix_to_angle_axis are the same, output and output_back are the same as well. 

So it seems that kornia.rotation_matrix_to_angle_axis is correct, while kornia.angle_axis_to_rotation_matrix is not.

**Comment by cvfish** [comment]
Further inverstigation seems that kornia.angle_axis_to_rotation_matrix is correct, while kornia.rotation_matrix_to_angle_axis is wrong.

**Comment by edgarriba** [comment]
@cvfish pip version of kornia is outdated. Have you tried with master ? We plan a release within the next couple of weeks with a lot of fixes since last version. Let us know, otherwise we can investigate whether there's a bug in that function.

**Comment by cvfish** [comment]
I actually tried the pip install from source using command 

pip install git+https://github.com/arraiyopensource/kornia

Is this version up to date ?

**Comment by edgarriba** [comment]
@cvfish installing directly with pip+git gives you the version in master. Have you found any workaround for this issue ? an alternative could be comparing against tf.graphics that implements same features https://github.com/tensorflow/graphics/blob/master/tensorflow_graphics/geometry/transformation/axis_angle.py#L147

**Comment by edgarriba** [comment]
@cvfish Any idea where the exact issue might be ?

**Comment by amonszpart** [comment]
Hi,

Great library!
I think I have a culprit, it's [`angle_axis_to_quaternion`](https://kornia.readthedocs.io/en/latest/_modules/kornia/geometry/conversions.html#angle_axis_to_quaternion) returning (w, x, y, z) instead of (x, y, z, w):

```python
def angle_axis_to_quaternion(angle_axis: torch.Tensor) -> torch.Tensor:
    r"""Convert an angle axis to a quaternion.
    The quaternion vector has components in (x, y, z, w) format.
    # ...
    return torch.cat([w, quaternion], dim=-1)
```

it should be 
`return torch.cat([quaternion, w], dim=-1)` I think.

Cheers,
Aron

**Comment by amonszpart** [comment]
Also the other way seems to also use (w, x, y, z) as input:
https://github.com/kornia/kornia/blob/b5418d8bac9e7bb01e66310d7d2b4227766c5a02/kornia/geometry/conversions.py#L529

shouldn't it be 
```python
    q1: torch.Tensor = quaternion[..., 0]
    q2: torch.Tensor = quaternion[..., 1]
    q3: torch.Tensor = quaternion[..., 2]
```?

**Comment by edgarriba** [comment]
@amonszpart agree - as you might noticed this module needs a bit of work yet

**Comment by remmel** [comment]
`kornia.angle_axis_to_rotation_matrix` is now `kornia.geometry.angle_axis_to_rotation_matrix`
